src/features/semanticTokensProvider.ts

```diff
@@ -187,7 +187,11 @@ export default class SemanticTokensProvider extends AbstractProvider implements
             return null;
         }
 
-        let req = createRequest<protocol.V2.SemanticHighlightRequest>(document, new vscode.Position(0, 0));
+        const currentDocument = vscode.workspace.textDocuments.find(d => d.fileName === document.fileName);
+        const isCurrentVersion = currentDocument?.version === document.version;
+
+        // We need to include the document contents in our request when we are highlighting a version of the document other than the current version, such as in the Diff view.
+        let req = createRequest<protocol.V2.SemanticHighlightRequest>(document, new vscode.Position(0, 0), /* includeBuffer: */ !isCurrentVersion);
```

Comment:
How does this work? AFAICT, the service itself actually doesn't look at the text in the buffer: https://github.com/OmniSharp/omnisharp-roslyn/blob/ff21ac475c/src/OmniSharp.Roslyn.CSharp/Services/SemanticHighlight/SemanticHighlightService.cs#L27

Reply:
@333fred Updated to use new omnisharp request property instead of setting the buffer.